google/cloud/pubsub/internal/tracing_batch_callback_test.cc

```diff
@@ -101,6 +103,83 @@ TEST(TracingBatchCallback, StartAndEndMessage) {
                             SpanHasEvents(EventNamed("gl-cpp.ack_end")))));
 }
 
+TEST(TracingBatchCallback, SubscribeAttributes) {
+  namespace sc = ::opentelemetry::trace::SemanticConventions;
+  auto span_catcher = InstallSpanCatcher();
+  auto mock = std::make_shared<pubsub_testing::MockBatchCallback>();
+  EXPECT_CALL(*mock, callback).Times(1);
+  auto batch_callback = MakeTestBatchCallback(std::move(mock));
+
+  batch_callback->callback(MakeResponse(1));
+  batch_callback->AckEnd("ack-id-0");
+
+  auto spans = span_catcher->GetSpans();
+  EXPECT_THAT(spans,
+              Contains(AllOf(SpanNamed("test-sub subscribe"),
+                             SpanHasAttributes(OTelAttribute<std::string>(
+                                 sc::kMessagingSystem, "gcp_pubsub")))));
+  EXPECT_THAT(spans,
+              Contains(AllOf(SpanNamed("test-sub subscribe"),
+                             SpanHasAttributes(OTelAttribute<std::string>(
+                                 "gcp.project_id", "test-project")))));
+  EXPECT_THAT(spans,
+              Contains(AllOf(SpanNamed("test-sub subscribe"),
+                             SpanHasAttributes(OTelAttribute<std::string>(
+                                 sc::kMessagingOperation, "subscribe")))));
+  EXPECT_THAT(spans,
+              Contains(AllOf(SpanNamed("test-sub subscribe"),
+                             SpanHasAttributes(OTelAttribute<std::string>(
+                                 sc::kMessagingMessageId, "id-0")))));
+  EXPECT_THAT(
+      spans,
+      Contains(AllOf(SpanNamed("test-sub subscribe"),
+                     SpanHasAttributes(OTelAttribute<std::string>(
+                         "messaging.gcp_pubsub.message.ack_id", "ack-id-0")))));
+  EXPECT_THAT(spans,
+              Contains(AllOf(SpanNamed("test-sub subscribe"),
+                             SpanHasAttributes(OTelAttribute<int64_t>(
+                                 "messaging.message.envelope.size", 101)))));
+  EXPECT_THAT(spans,
+              Contains(AllOf(SpanNamed("test-sub subscribe"),
+                             SpanHasAttributes(OTelAttribute<std::string>(
+                                 sc::kMessagingDestinationName, "test-sub")))));
```

Comment:
It looks like all of this could be simplified to ...
```
  EXPECT_THAT(
      spans,
      Contains(AllOf(
          SpanNamed("test-sub subscribe"),
          SpanHasAttributes(
              OTelAttribute<std::string>(sc::kMessagingSystem, "gcp_pubsub"),
              OTelAttribute<std::string>("gcp.project_id", "test-project"),
              OTelAttribute<std::string>(sc::kMessagingOperation, "subscribe"),
              OTelAttribute<std::string>(sc::kMessagingMessageId, "id-0"),
              OTelAttribute<std::string>("messaging.gcp_pubsub.message.ack_id", "ack-id-0"),                             
              OTelAttribute<int64_t>("messaging.message.envelope.size", 101),
              OTelAttribute<std::string>(sc::kMessagingDestinationName, "test-sub")))));
```

Reply:
Done